swarm/storage/resource.go

```diff
@@ -760,8 +770,8 @@ func (self *ResourceHandler) parseUpdate(chunkdata []byte) (*Signature, uint32,
 // A resource update cannot span chunks, and thus has max length 4096
 func (self *ResourceHandler) UpdateMultihash(ctx context.Context, name string, data []byte) (Key, error) {
 	// \TODO perhaps this check should be in newUpdateChunk()
-	if isMultihash(data) == 0 {
-		return nil, NewResourceError(ErrNothingToReturn, "Invalid multihash")
+	if _, _, err := swarmhash.GetMultihashLength(data); err != nil {
+		return nil, NewResourceError(ErrNothingToReturn, fmt.Sprintf("Invalid multihash: %v", err))
```

Comment:
this error is not so crystal clear, why can't you just bubble the error from `GetMultihashLength` up?

Reply:
the error texts in `swarm/hash` are pretty neutral, e.g "unreadable length field", not so easy to tell further up what that means.

Do you mean we should have typed error maybe?